### PR TITLE
change testing from unittest to pytest

### PR DESCRIPTION
**Pull request by seeni-dev**
1. fix tox.ini and test_datastore.py(py-datastore/datastore/filesystem/test_filesystem.py)

### CODE REVIEW

**Comment by seeni-dev** [comment]
To test this PR, use ` pytest -v  -k "test_filesystem.py" `

**Comment by ntninja** [comment]
Is this PR ready for reviewing at this point in time?

I don't know if you're still planning on it, but in the longer run, I was hoping that we could convert all tests to use pytest [assertions](https://docs.pytest.org/en/latest/assert.html) and [fixtures](https://docs.pytest.org/en/latest/fixture.html). I must admit that it doesn't have much priority right now, but maybe keep that in mind when updating stuff.

I also see you've gotten the number of test errors down, so thanks for this already.

File: tox.ini [hunk 1]
```diff
@@ -29,7 +29,7 @@ max-line-length = 120
 
 [pytest]
 python_files =
-;    test_*.py
+    test_*.py
```
Comment (AliabbasMerchant) [inline tox.ini:32]: @Seenivasanseeni I had commented this out, so that we can focus on 1 particular module at one time. But nvm. 👍

**Comment by seeni-dev** [comment]
@alexander255  Can you review this PR?.

**Comment by seeni-dev** [comment]
To test `pytest datastore/filesystem/test_filesystem.py`

File: datastore/core/test/test_basic.py [hunk 2]
```diff
@@ -93,9 +93,9 @@ def subtest_queries(self):
 
         self.check_query(Query(k), n, slice(0, n))
         self.check_query(Query(k, limit=n), n, slice(0, n))
-        self.check_query(Query(k, limit=n / 2), n, slice(0, n / 2))
-        self.check_query(Query(k, offset=n / 2), n, slice(n / 2, n))
-        self.check_query(Query(k, offset=n / 3, limit=n / 3), n, slice(n / 3, 2 * (n / 3)))
+        self.check_query(Query(k, limit=n // 2), n, slice(0, n // 2))
+        self.check_query(Query(k, offset=n // 2), n, slice(n // 2, n))
+        self.check_query(Query(k, offset=n // 3, limit=n // 3), n, slice(n // 3, 2 * (n // 3)))
```
Comment (ntninja) [inline datastore/core/test/test_basic.py:98]: Nice :+1:

File: datastore/filesystem/test_filesystem.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import os
+import shutil
+import pytest
```
Comment (ntninja) [inline datastore/filesystem/test_filesystem.py]: Why import the `pytest` module here if you don't use it?

File: datastore/filesystem/test_filesystem.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        if os.path.exists(self.tmp):
+            shutil.rmtree(self.tmp)
+
+    def test_datastore(self):
+        dirs = map(str, range(0, 4))
+        dirs = map(lambda d: os.path.join(self.tmp, d), dirs)
+        fses = map(FileSystemDatastore, dirs)
+        dses = list(map(serialize.shim, fses))
+        self.subtest_simple(dses, numelems=500)
+
```
Comment (ntninja) [inline datastore/filesystem/test_filesystem.py]: Did you maybe forget to commit the deletion of `test.py` here?

**Review by ntninja — COMMENTED** [review]
One tiny thing, otherwise LGTM

File: datastore/filesystem/test_filesystem.py (hunk 4 above)
Comment (seeni-dev) [reply datastore/filesystem/test_filesystem.py]: that was intentional. but i will remove it anyway.

**Comment by seeni-dev** [comment]
is the PR good now? @alexander255

**Review by ntninja — APPROVED** [review]
Thanks, it's easier to read now! :+1:

(See how git treats the create + remove as a move in the diff view, so that you can see the actual changes that happened between the two files? That's why I asked about this in the first place.)

**Pull request merged**